server/task_executor_schema_update_ghost_sync.go

```diff
@@ -1,26 +1,232 @@
+		return nil, err
+	}
+	return migrationContext, nil
 }
 
 // RunOnce will run SchemaUpdateGhostSync task once.
 func (exec *SchemaUpdateGhostSyncTaskExecutor) RunOnce(ctx context.Context, server *Server, task *api.Task) (terminated bool, result *api.TaskRunResultPayload, err error) {
-	return true, nil, fmt.Errorf("not implemented yet")
+	defer func() {
+		if r := recover(); r != nil {
+			panicErr, ok := r.(error)
+			if !ok {
+				panicErr = fmt.Errorf("%v", r)
+			}
+			exec.l.Error("SchemaUpdateGhostSyncTaskExecutor PANIC RECOVER", zap.Error(panicErr), zap.Stack("stack"))
+			terminated = true
+			err = fmt.Errorf("encounter internal error when executing migration using gh-ost")
+		}
+	}()
+
+	payload := &api.TaskDatabaseSchemaUpdateGhostSyncPayload{}
+	if err := json.Unmarshal([]byte(task.Payload), payload); err != nil {
+		return true, nil, fmt.Errorf("invalid database schema update gh-ost sync payload: %w", err)
+	}
+	return runGhostMigration(ctx, exec.l, server, task, db.Migrate, payload.Statement, payload.SchemaVersion, payload.VCSPushEvent)
+}
+
+func runGhostMigration(ctx context.Context, l *zap.Logger, server *Server, task *api.Task, migrationType db.MigrationType, statement, schemaVersion string, vcsPushEvent *vcsPlugin.PushEvent) (terminated bool, result *api.TaskRunResultPayload, err error) {
+	mi, err := preMigration(ctx, l, server, task, migrationType, statement, schemaVersion, vcsPushEvent)
+	if err != nil {
+		return true, nil, err
+	}
+	migrationID, schema, err := executeSync(ctx, l, task, mi, statement)
+	if err != nil {
+		return true, nil, err
+	}
+	return postMigration(ctx, l, server, task, vcsPushEvent, mi, migrationID, schema)
+}
+
+func executeSync(ctx context.Context, l *zap.Logger, task *api.Task, mi *db.MigrationInfo, statement string) (int64, string, error) {
+	statement = strings.TrimSpace(statement)
+
+	driver, err := getAdminDatabaseDriver(ctx, task.Instance, task.Database.Name, l)
+	if err != nil {
+		return -1, "", err
+	}
+	defer driver.Close(ctx)
+	setup, err := driver.NeedsSetupMigration(ctx)
+	if err != nil {
+		return -1, "", fmt.Errorf("failed to check migration setup for instance %q: %w", task.Instance.Name, err)
+	}
+	if setup {
+		return -1, "", common.Errorf(common.MigrationSchemaMissing, fmt.Errorf("missing migration schema for instance %q", task.Instance.Name))
+	}
+
+	executor := driver.(util.MigrationExecutor)
+
+	var prevSchemaBuf bytes.Buffer
+	if err := driver.Dump(ctx, mi.Database, &prevSchemaBuf, true); err != nil {
+		return -1, "", err
+	}
+
+	insertedID, err := util.BeginMigration(ctx, executor, mi, prevSchemaBuf.String(), statement, db.BytebaseDatabase)
+	if err != nil {
+		return -1, "", err
+	}
+	startedNs := time.Now().UnixNano()
+
+	err = executeGhost(task.Instance, task.Database.Name, statement)
+	if err != nil {
+		return -1, "", err
+	}
+
+	var afterSchemaBuf bytes.Buffer
+	if err := executor.Dump(ctx, mi.Database, &afterSchemaBuf, true); err != nil {
```

Comment:
Should executor.Dump() be part of EndMigration() because that's a required step for migration?

Note: we can do refactoring after this PR though.

Reply:
SGTM. I'll refactor after this PR.